app/client/src/components/shared/AddSaveDataWidget.SavePanel.tsx

```diff
@@ -0,0 +1,722 @@
+
+  // Watch for when layers are added to the map
+  const [mapLayerCount, setMapLayerCount] = useState(mapView.map.layers.length);
+  const [layerWatcher, setLayerWatcher] = useState<IHandle | null>(null);
+  useEffect(() => {
+    if (!mapView || layerWatcher) return;
+
+    const watcher = reactiveUtils.watch(
+      () => mapView.map.layers.length,
+      () => setMapLayerCount(mapView.map.layers.length),
+    );
+
+    const layerWatchers: { [id: string]: IHandle } = {};
+
+    function watchLayerVisibility(layer: __esri.Layer) {
+      const visibilityWatcher = reactiveUtils.watch(
+        () => layer.visible,
+        () => {
+          if (
+            layersToIgnore.includes(layer.id) ||
+            layerTypesToIgnore.includes(layer.type)
+          )
+            return;
+
+          setSaveLayersList((layersList) => {
+            const newLayersList = { ...layersList };
+            newLayersList[layer.id].toggled = layer.visible;
+            return newLayersList;
+          });
+        },
+      );
+      layerWatchers[layer.id] = visibilityWatcher;
+    }
+
+    function watchLayerLoaded(layer: __esri.Layer) {
+      const loadedWatcher = reactiveUtils.watch(
+        () => layer.loaded,
+        () => {
+          if (!layer.loaded) return;
+
+          loadedWatcher.remove();
+
+          setSaveLayersList((layersList) => {
+            const newLayersList = { ...layersList };
+            if (newLayersList[layer.id])
+              newLayersList[layer.id].label = layer.title;
+            return newLayersList;
+          });
+        },
+      );
+    }
+
+    mapView.map.layers.forEach((layer) => watchLayerVisibility(layer));
+
+    mapView.map.layers.on('change', (e) => {
+      if (e.added.length > 0) {
+        e.added.forEach((layer) => {
+          watchLayerVisibility(layer);
+          if (!layer.loaded) watchLayerLoaded(layer);
+        });
+      }
+
+      if (e.removed.length > 0) {
+        e.removed.forEach((layer) => {
+          layerWatchers[layer.id].remove();
+          delete layerWatchers[layer.id];
+
+          setSaveLayersList((layersList) => {
+            const newLayersList = { ...layersList };
+            delete newLayersList[layer.id];
+            return newLayersList;
+          });
+        });
+      }
+    });
+
+    setLayerWatcher(watcher);
+  }, [layerWatcher, mapView, setSaveLayersList]);
+
+  const [layersInitialized, setLayersInitialized] = useState(false);
+  useEffect(() => {
+    if (layersInitialized) return;
+
+    if (saveLayersList) {
+      // build object of switches based on layers on map
+      const newSwitches = { ...saveLayersList };
+      mapView.map.layers.forEach((layer) => {
+        if (
+          layersToIgnore.includes(layer.id) ||
+          layerTypesToIgnore.includes(layer.type)
+        )
+          return;
+
+        if (newSwitches.hasOwnProperty(layer.id)) {
+          newSwitches[layer.id].layer = layer;
+        }
+      });
```

Comment:
Currently, this block doesn't change state. Is `saveLayersList` supposed to be updated here?
```suggestion
      });
      setSaveLayersList(newSwitches);
```

Reply:
Good catch! That entire `useEffect` is unnecessary, so I'm going to pull that code out. I think that was some code that was mainly for testing, that I just forgot to remove.